### DISCUSSION TITLE
[Bug] iOS Tab Page Inconsistent Behavior

### DISCUSSION THREAD

**Discussion by albilaga** [xamarin/Xamarin.Forms#10630] [General]
### Description
iOS tab page when clicking active tab not behaving the same as in Android or in native iOS (using SwiftUI)

### Steps to Reproduce
1. When I am on the tabbed page (let's say I am on the first tab page A), I navigate to another page (still in the 1st tab, and the 1st tab page using `NavigationPage`).
2. Now I am on the page B in first tab.
3. Then I click on 1st tab icon. 

### Expected Behavior
The tab should do nothing and the page not getting reset.

### Actual Behavior
The page getting reset back to root page. I test it in Android and using SwiftUI on iOS this is not happened.

### Basic Information

- Version with issue: 4.6.0.726
- Last known good version: -
- IDE: Visual Studio for Mac
- Platform Target Frameworks: 
  - iOS: 13.4
  - Android: 9.0

### Videos
I attached the [videos](https://github.com/xamarin/Xamarin.Forms/files/4603742/Videos.Tab.Page.zip) in here with result from iOS using SwiftUI, Android and iOS using Xamarin Forms.

### Reproduction Link
This is the [sample](https://github.com/xamarin/Xamarin.Forms/files/4603749/tabtest.zip) using Xamarin Forms
This is the [sample](https://github.com/xamarin/Xamarin.Forms/files/4603755/tabbedtestiOS.zip) using SwiftUI iOS

**Comment by albilaga** [comment]
Looks like SwiftUI sample is not the same as using `NavigationController` so maybe the sample is irrelevant. 
[Here](https://github.com/xamarin/Xamarin.Forms/files/4605822/TestTabiOSNative.zip) I attach another sample using Storyboard and `UINavigationController` to embed first tab and the result the same as in Xamarin Forms (when active tab get clicked again, it will reset to root page). But it will be good if we have consistent behavior between android and iOS

**Comment by PureWeen** [comment]
I'm thinking a property would be best for this on the TabView

That way users could indicate if they want double tap to PopToRoot or not. Then they could configure it to work one way vs the other. 

We can't really change how iOS works as that would be a fairly heavy breaking change plus iOS users kind of expect this type of behavior.

We don't implement anything on iOS to enable this behavior it's just iOS being iOS

**Comment by jsuarezruiz** [comment]
@albilaga What do you think about Shane's last comment?, does it seem like an option to allow both behaviors in a simple way?

**Comment by albilaga** [comment]
@PureWeen @jsuarezruiz Yes I think it will be better if we have the property to avoid breaking change. There are some people that also will want to expect current behavior but with user also become platform agnostic there are also some people (including me) that want behavior to be consistent in all platform

**Comment by jsuarezruiz** [comment]
I get it. What do you think if you create an enhancement with your proposal for the new property? I can create it myself if you want.

**Comment by albilaga** [comment]
@jsuarezruiz what do you think the best property name for it? I think we can put it in iOS specific for `TabbedPage` but I am not sure with the property name. I am thinking it we can use bool for value. `IsBarItemResetPage` is too verbose or not?
